temporal-sdk/src/test/java/io/temporal/workflow/shared/TestWorkflowWithCronScheduleImpl.java

```diff
@@ -46,15 +49,16 @@ public String execute(String testName) {
       return null;
     }
 
-    lastCompletionResult = Workflow.getLastCompletionResult(String.class);
     lastFail = Workflow.getPreviousRunFailure();
+    retryCount.computeIfAbsent(testName, k -> new AtomicInteger());
+    lastCompletionResults.computeIfAbsent(testName, k -> new HashMap<>());
 
-    AtomicInteger count = retryCount.get(testName);
-    if (count == null) {
-      count = new AtomicInteger();
-      retryCount.put(testName, count);
+    int c = retryCount.get(testName).incrementAndGet();
+    String result = Workflow.getLastCompletionResult(String.class);
+    if (result != null) {
+      lastCompletionResult.put(testName, Workflow.getLastCompletionResult(String.class));
     }
-    int c = count.incrementAndGet();
+    lastCompletionResults.get(testName).put(c, lastCompletionResult.get(testName));
```

Comment:
Why do you need to make a `lastCompletionResult.get(testName)` here if you just published this value? Why not to use `result` directly?